nav2_costmap_2d/plugins/costmap_filters/binary_filter.cpp

```diff
@@ -259,6 +295,61 @@ void BinaryFilter::changeState(const bool state)
     std::make_unique<std_msgs::msg::Bool>();
   msg->data = state;
   binary_state_pub_->publish(std::move(msg));
+  changeParameters(state);
+}
+
+void BinaryFilter::changeParameters(const bool state)
+{
+  for (size_t param_index = 0; param_index < binary_parameters_info_.size(); ++param_index) {
+    if (!change_parameters_clients_.at(param_index)->wait_for_service(
+        std::chrono::milliseconds(change_parameter_timeout_)))
+    {
+      RCLCPP_WARN(
+        logger_, "BinaryFilter:  service %s not available. Skipping ...",
+        change_parameters_clients_.at(param_index)->get_service_name());
+      continue;
+    } else {
+      RCLCPP_INFO(
+        logger_, "BinaryFilter:  service %s available.",
+        change_parameters_clients_.at(param_index)->get_service_name());
+    }
+
+    // Create a rcl_interfaces::msg::SetParameters client for changing parameters
+    auto request = std::make_shared<rcl_interfaces::srv::SetParameters::Request>();
+
+    // Set parameters for BinaryFilter
+    rcl_interfaces::msg::Parameter bool_param;
+    bool_param.name = binary_parameters_info_.at(param_index).param_name;
+    bool_param.value.type = rcl_interfaces::msg::ParameterType::PARAMETER_BOOL;
+    if (binary_parameters_info_.at(param_index).default_state != default_state_) {
```

Comment:
This way is some kind of tricky: the state of each parameter will flip in the same or opposite way depending on whether the parameter's initial default state will be equal to global filter state. So, this construction will work, but not it is obvious.

How about to change the logic a bit to express the intention more clearly:
```
    const bool param_default_state = binary_parameters_info_.at(param_index).default_state;
    if (state == default_state_) {
      // Filter is not flipped
      bool_param.value.bool_value = param_default_state;
    } else {
      // Filter is flipped
      bool_param.value.bool_value = !param_default_state;
    }
```

Reply:
> How can you have a global default_state_ for potentially many unrelated parameters in unrelated nodes?

I think it's realistic to have heavy relations between parameters, given that the mask is shared. For instance one could set it to disable an instance of the obstacle layer and enable another one (and this may hold for other layers too). This way one would need to specify only the default value for half of the parameter, while the other half can be left default (as the global default state)

Reply:
> This way is some kind of tricky: the state of each parameter will flip in the same or opposite way depending on whether the parameter's initial default state will be equal to global filter state. So, this construction will work, but not it is obvious.
> 
> How about to change the logic a bit to express the intention more clearly:
> ...

Thank you, much more readable!